scripts/Jenkinsfile

```diff
@@ -0,0 +1,164 @@
+    body: "${status}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]'; see console output at ${env.BUILD_URL}"
+  )
+}
+
+node ('upstream-merger')   
+{
+  try { 
+	     def dateFormat = new SimpleDateFormat("yyyyMMddHH")
+		 def date = new Date()
+		 def nproc = 8
+	     String userID = "rocm-hcc"
+	     String merge_name = "auto_merge_" + dateFormat.format(date)
+	     String user_email = "rocm.hcc@amd.com"
+     
+        def startStage = params.START_STAGE.toInteger()
+        
+        if (startStage <= 1) { 
+			stage ('Stage 1: Clone HCC with repo') 
+			{ 
+			    deleteDir()
+				sh (script: "git clone git@github.com:rocm-hcc/hcc.git hcc_fork")
+				dir("hcc_fork") {
+					sh (script: "git checkout clang_tot_upgrade")
+					sh (script: "git submodule update --init")
+				}
+			}
+		}
+				
+		if (startStage <= 2) { 
+			stage ('Stage 2: Add all remotes and forks for submodules') {
+				git_remote_add("hcc_fork","hcc_fork", "ssh://git@github.com/${ userID}/hcc.git")
+				git_remote_add("hcc_fork/lld","lld_fork", "ssh://git@github.com/${ userID}/lld.git")
+				git_remote_add("hcc_fork/compiler","llvm_fork", "ssh://git@github.com/${ userID}/llvm.git")
+				git_remote_add("hcc_fork/compiler-rt","compiler-rt_fork", "ssh://git@github.com/${ userID}/compiler-rt.git")
+				git_remote_add("hcc_fork/compiler-rt","compiler-rt", "ssh://git@github.com/llvm-mirror/compiler-rt.git")
+				git_remote_add("hcc_fork/clang-tools-extra","clang-tools-extra_fork", "ssh://git@github.com/${ userID}/clang-tools-extra.git")
+				git_remote_add("hcc_fork/clang-tools-extra","clang-tools-extra", "ssh://git@github.com/llvm-mirror/clang-tools-extra.git")
+				git_remote_add("hcc_fork/clang","clang_fork", "ssh://git@github.com/${ userID}/hcc-clang-upgrade.git")
+				git_remote_add("hcc_fork/clang","clang", "ssh://git@github.com/RadeonOpenCompute/clang")
+			}
+		}
+		
+		if (startStage <= 3) { 
+			stage ('Stage 3:Checkout latest master ROCM-Device-Libs') {
+				sh (script: "cd hcc_fork/rocdl; git fetch --all; git checkout origin/master")
+			}
+		}
+		
+		if (startStage <= 4) { 		
+			stage ('Stage 4: Merge amd-common  commits') {
+			    
+			    if (!fileExists("merge-branch-name") ) { 
+			    	writeFile file: "merge-branch-name", text: "${ merge_name}"
+				} 
+				merge_name = readFile "merge-branch-name"
+				
+				dir("hcc_fork") {
+					git_config()
+					git_merge("lld", merge_name, "origin/amd-hcc", "origin/amd-common")
+					git_merge("compiler",merge_name,"origin/amd-hcc", "origin/amd-common")
+					git_merge("compiler-rt", merge_name, "origin/amd-hcc", "--no-ff compiler-rt/master")
+					git_merge("clang-tools-extra", merge_name, "origin/amd-hcc", "clang-tools-extra/master")
+					git_merge("clang", merge_name, "origin/clang_tot_upgrade", "--no-ff clang/amd-common")
+				}
+			}		
+		}
+		
+		if (startStage <= 5) { 
+			stage ('Stage 5: Build merged HCC') {
+				sh (script: "mkdir -p build; cd build; cmake -DCMAKE_BUILD_TYPE=Release ../hcc; make -j${ nproc}")
```

Comment:
@david-salinas is it possible to do 
```
make -j$(nproc)
```
here? so `nproc` is executed to help determine available CPU cores. right now it seems to be fixed at 8 in jenkins file.